apis/mutations/login/useLogin.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import { useMutation } from '@tanstack/react-query';
+import { signIn } from 'next-auth/react';
+import { useRouter } from 'next/navigation';
+import { toast } from 'react-toastify';
+import { AxiosError } from 'axios';
+
+export type LoginParams = {
+  email: string;
+  password: string;
+  cfTurnstileResponse?: string;
+  userType: string;
+  redirectUrl: string;
+  remember?: boolean;
+};
+
+type ErrorResponse = {
+  message: string;
+  statusCode?: number;
+};
+
+const useLogin = () => {
+  const router = useRouter();
+
+  return useMutation({
+    mutationFn: async (
+      formValues: LoginParams & { userType: string; redirectUrl: string; remember: boolean }
+    ) => {
+      const { email, password, cfTurnstileResponse, userType, redirectUrl, remember } = formValues;
+
+      const result = await signIn('credentials', {
+        redirect: false,
+        email,
+        password,
+        cfTurnstileResponse,
+        userType,
+        remember,
+      });
+
+      if (!result?.ok) {
+        throw new Error(result?.error || 'Неуспешна најава');
+      }
+
+      router.push(redirectUrl);
```

Comment:
It would be better to not reroute inside the mutation. Because of the asynchronous nature of mutations here is not the best place to handle side effects.
We can reroute inside the component , when we use the mutation, based on the mutation state.

Reply:
Good point. Thanks for the feedback!